pkg/api/watcher_gateway.go

```diff
@@ -917,6 +913,33 @@ func (w *WatcherGateway) upsertIngressesOnNamespace(ctx context.Context, namespa
 	return nil
 }
 
+func (w *WatcherGateway) listGatewayIngresses(namespace, gatewayName string) ([]*netv1.Ingress, error) {
+	managedByHub, err := labels.NewRequirement("app.kubernetes.io/managed-by", selection.Equals, []string{"traefik-hub"})
+	if err != nil {
+		return nil, fmt.Errorf("create managed by hub requirement: %w", err)
+	}
+	hubIngressesSelector := labels.NewSelector().Add(*managedByHub)
+
+	hubIngresses, err := w.kubeInformer.Networking().V1().Ingresses().Lister().Ingresses(namespace).List(hubIngressesSelector)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list ingresses: %w", err)
+	}
+
+	ingressName, err := getIngressName(gatewayName)
+	if err != nil {
+		return nil, fmt.Errorf("get ingress name for hub domain: %w", err)
+	}
+
+	gatewayIngresses := hubIngresses[:0]
```

Comment:
Out of curiosity, why not using?
```suggestion
	var gatewayIngresses []*netv1.Ingress
```